src/script/descriptor.cpp

```diff
@@ -352,7 +352,7 @@ class ConstPubkeyProvider final : public PubkeyProvider
     }
 };
 
-enum class DeriveType {
+enum class RangedDeriveType {
     NO,
     UNHARDENED,
     HARDENED,
```

Comment:
Agree that based on the `IsRange()` implementation, this enum could be clearer. But would lean towards only changing the values, not the name in this case:
```suggestion
enum class DeriveType {
    SINGLE,
    UNHARDENED_RANGE,
    HARDENED_RANGE,
```

Reply:
Accepted the suggestion with some modification by using `NON_RANGED` instead of `SINGLE`, which felt incorrect to me. I do like reading `DeriveType::NON_RANGED` instead of the original `DeriveType::NO`, which made me feel like there was no child derivation at all and prompted me to raise the PR.

I had to experiment with the `sed` command in the scripted diff a bit to get the right order working but it was fun.